### ISSUE TITLE
Password changer enforces obsolete password requirement

### ISSUE THREAD

**Issue by strypey** [dreamwidth/dw-free#2392]
My current passphrase practice is to follow the advice in [XKCD #936](https://explainxkcd.com/wiki/index.php/936:_Password_Strength/). Unfortunately, I keep striking sites whose password changing system a) wants to do my thinking for me, and b) is stuck in the 1990s, believing that "Tr0ub4dor&3" is a more secure password than "correcthorsebatterystaple". I just tried to change my passphrase, and sadly, it turns out Dreamwidth is one of them. It refused to accept my new password, with this error message:

> Bad new password: Your password must contain at least one non-letter character (a digit or symbol).

 Two suggestions:
* if you agree with XKCD that requiring a non-letter character doesn't makes passphrases stronger, remove the code that enforces this from the password changer (also consider changing all mentions of "password" to "passphrase").
* if you disagree with XKCD, give the advice about using a non-letter character to the user, but allow them to set a passphrase that doesn't follow that schema anyway.

**Comment by momijizukamori** [comment]
Yeah, research has shown that enforcing higher minimum lengths leads to better passwords than enforcing complexity requirements. I forget what we've got as minimum right now, but NIST recommends at least 8, and OWASP recommends including visual strength meter and checking against passwords that appear in common breaches.

- [OWASP recommendations](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html)
- [NIST guidelines](https://pages.nist.gov/800-63-3/sp800-63b.html)
- [Summary of NIST guidelines](https://blog.netwrix.com/2022/11/14/nist-password-guidelines/)
